javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -246,6 +246,34 @@ void setCurrentArray() {
         };
     }
 
+    @SuppressWarnings("unchecked")
+    <T2> void visit(LeafVisitor<T2[]> visitor) {
+        int start = lastDigit(offset);
+        for (int index = 0; index < length; ) {
+            final T2[] leaf = (T2[]) getLeaf(index);
+            final int end = Math.min(leaf.length, start + length - index);
+
+            visitor.visit(leaf, start, end);
+
+            index += end - start;
+            start = 0;
+        }
+    }
+
+    <U> BitMappedTrie<U> map(Function<? super T, ? extends U> mapper) {
```

Comment:
Could also unroll the middle manually, resulting in a measurable change, but distorting the code significantly, i.e.
```java
int i = start;
for (; i < end - 3; i += 4) {
    final int val = index.val;
    elems[val] = mapper.apply(leaf[i]);
    elems[val + 1] = mapper.apply(leaf[i + 1]);
    elems[val + 2] = mapper.apply(leaf[i + 2]);
    elems[val + 3] = mapper.apply(leaf[i + 3]);
    index.val = val + 4;
}
for (; i < end; i++) {
    elems[index.val++] = mapper.apply(leaf[i]);
}
```
before:
```java
Map slang_persistent/scala_persistent 1.26×  1.53×  1.88×
```

after:
```java
Map slang_persistent/scala_persistent 1.40×  1.54×  1.92×
```

i.e. not worth it